python/pyarrow/io.pxi

```diff
@@ -678,6 +678,9 @@ cdef class Buffer:
         else:
             return NotImplemented
 
+    def __reduce__(self):
+        return py_buffer, (self.to_pybytes(),)
```

Comment:
Mmmh... if you're pickling a mutable buffer, it will give you a back a non-mutable buffer (since you are reducing it as a bytes object). Is it what you expect?

Reply:
Yes, this is what I would expect for now. Data pickled and unpickled again is not anymore in the environment that created it. Thus I would disallow mutations for now.

Once we also have mutable buffers and have a use case for unpickling them as such later on, we should add a specialization that takes care off also passing this attribute.